packages/client/src/lib/reform/ReForm.re

```diff
@@ -0,0 +1,290 @@
+module Helpers = ReForm_Helpers;
+
+module Validation = ReForm_Validation;
+
+module Value = ReForm_Value;
+
+/* Validation types */
+let safeHd = lst => List.length(lst) == 0 ? None : Some(List.hd(lst));
+
+let (>>=) = (value, map) =>
+  switch (value) {
+  | None => None
+  | Some(value) => map(value)
+  };
+
+module type Config = {
+  type state;
+  type fields;
+  /* (fieldName, getter, setter) */
+  let lens: list((fields, state => Value.t, (state, Value.t) => state));
+};
+
+module Create = (Config: Config) => {
```

Comment:
Did you copy this from the ReForm codebase? Do you think there could be a more elegant way, like just having the jsx3 part?

Reply:
Yes, I have copied the lib and added Jsx3. But now I realized that may be we can just add dedicated Jsx3 "shells" for these Jsx2 components.

Reply:
Makes sense. The Jsx3 shells sound good to me.